src/java/org/apache/cassandra/io/sstable/Component.java

```diff
@@ -172,6 +175,20 @@ public File getFile(String absolutePath)
         return StorageProvider.instance.withOpenOptions(ret, this);
     }
 
+    /**
+     * @return file that uses the same file system as provided directory
+     */
+    public File getFile(File directory, String filenamePart)
```

Comment:
please add unit tests about this method, it is easy to forget edge cases or introduce regressions in the future

we can probably have a static method that does the path handling, so that doing unit tests is easier

Reply:
added basic unit test in cc. I will add more in CNDB side with remote file system